dask_cloudprovider/providers/azure/azureml.py

```diff
@@ -0,0 +1,769 @@
+
+        ### DATASTORES
+        self.datastores = datastores
+
+        ### FUTURE EXTENSIONS
+        self.kwargs = kwargs
+
+        ### RUNNING IN MATRIX OR LOCAL
+        self.same_vnet = None
+
+        ### GET RUNNING LOOP
+        self._loop_runner = LoopRunner(loop=None, asynchronous=asynchronous)
+        self.loop = self._loop_runner.loop
+
+        self.abs_path = pathlib.Path(__file__).parent.absolute()
+
+        ### INITIALIZE CLUSTER
+        super().__init__(asynchronous=asynchronous)
+
+        if not self.asynchronous:
+            self._loop_runner.start()
+            self.sync(self.get_defaults)
+            self.sync(self.create_cluster)
+
+    async def get_defaults(self):
+        self.config = dask.config.get("cloudprovider.azure", {})
+
+        if self.experiment_name is None:
+            self.experiment_name = self.config.get("experiment_name")
+
+        if self.initial_node_count is None:
+            self.initial_node_count = self.config.get("initial_node_count")
+
+        if self.jupyter is None:
+            self.jupyter = self.config.get("jupyter")
+
+        if self.jupyter_port is None:
+            self.jupyter_port = self.config.get("jupyter_port")
+
+        if self.dashboard_port is None:
+            self.dashboard_port = self.config.get("dashboard_port")
+
+        if self.scheduler_port is None:
+            self.scheduler_port = self.config.get("scheduler_port")
+
+        if self.additional_ports is None:
+            self.additional_ports = self.config.get("additional_ports")
+
+        if self.admin_username is None:
+            self.admin_username = self.config.get("admin_username")
+
+        if self.admin_ssh_key is None:
+            self.admin_ssh_key = self.config.get("admin_ssh_key")
+
+        if self.datastores is None:
+            self.datastores = self.config.get("datastores")
+
+        ### PARAMETERS TO START THE CLUSTER
+        self.scheduler_params = {}
+        self.worker_params = {}
+
+        ### scheduler and worker parameters
+        self.scheduler_params["--jupyter"] = True
+
+        if self.use_gpu:
+            self.scheduler_params["--use_gpu"] = True
+            self.scheduler_params["--n_gpus_per_node"] = self.n_gpus_per_node
+            self.worker_params["--use_gpu"] = True
+            self.worker_params["--n_gpus_per_node"] = self.n_gpus_per_node
+
+        ### CLUSTER PARAMS
+        self.max_nodes = self.compute_target.serialize()["properties"]["properties"][
+            "scaleSettings"
+        ]["maxNodeCount"]
+        self.scheduler_ip_port = None
+        self.workers_list = []
+        self.URLs = {}
+
+        ### SANITY CHECKS
+        ###-----> initial node count
+        if self.initial_node_count > self.max_nodes:
+            self.initial_node_count = self.max_nodes
+
+    def __print_message(self, msg, length=80, filler="#", pre_post=""):
+        logger.info(msg)
+        print(f"{pre_post} {msg} {pre_post}".center(length, filler))
```

Comment:
I think we need to drop this line now that we are using logging.

```suggestion
```

Reply:
We would still want to keep it. They way we currently instantiate the cluster is we submit an `Experiment Run` to AzureML service. It takes a bit to spin up (currently around 10 mins from cold start and ~1-2 minutes when the nodes are hot); we're working on bringing this down to 1 minute in the short term. Without any meaningful messages to the user, the experience would be severely impacted and might cause a lot of confusion. 

We'll most likely remove/replace it when the cluster spin up time is around 1 minute but until then -- we'd like to keep the `print(...)` statements. 

I will update the docs tonight and will provide better example how to 'get started'. For now I don't think we'll provide any CLI experience.

Here's how the experience looks for now: [GTC/SJ_2020/workshop/1_Setup/Setup.ipynb](https://github.com/drabastomek/GTC/blob/master/SJ_2020/workshop/1_Setup/Setup.ipynb)

Reply:
Ok sure that makes sense.

That notebook looks good. Just walking through each section of that with a bit of context about why and what it does would make a great documentation page.

Reply:
Thanks! I was thinking along the same lines wrt the documentation. I will try to finalize that tonight!

Reply:
@jacobtomlinson Finally got to the doc page. Let me know if any changes required! Thanks!